src/System.Device.Gpio.Tests/GpioControllerTestBase.cs

```diff
@@ -116,12 +116,12 @@ public void ThrowsIfWritingOnInputPin()
         }
 
         [Fact]
-        public void ThrowsIfReadingFromOutputPin()
+        public void CanReadFromOutputPin()
         {
             using (GpioController controller = new GpioController(GetTestNumberingScheme(), GetTestDriver()))
             {
                 controller.OpenPin(OutputPin, PinMode.Output);
-                Assert.Throws<InvalidOperationException>(() => controller.Read(OutputPin));
+                Assert.Equal(PinValue.Low, controller.Read(OutputPin));
```

Comment:
please also `gpioController.Write` and assert that value has changed afterwards and we can merge like that